arkouda/sorting.py

```diff
@@ -158,8 +158,9 @@ def coargsort(
             size = a.size
         elif size != a.size:
             raise ValueError("All pdarrays, Strings, or Categoricals must be of the same size")
+
     if size == 0:
-        return zeros(0, dtype=arrays[0].dtype)
+        return zeros(0, dtype=int if a.dtype == 'str' else arrays[0].dtype)
```

Comment:
We're referencing `a` which is defined in the `for` loop. That feels risky to me since it could be garbage collected. We should try and use `arrays[0]` instead

The other things is doing `== 'str'`, doing direct comparison against a hard coded value isn't best practices. I would recommend using `isinstance` instead for type comparisons

Overall I think changing this to something like
```python
return zeros(0, dtype=int if isinstance(arrays[0], (Strings, Categorical)) else arrays[0].dtype)
```
would be better